example: Add second, simpler example.

vi opened the pull request:
There is already "egui-graph-edit-example", but it may be too big and overwhelming to start from.

Here is an alternative example that may serve as a better starting point for learning egui-graph-edit and maybe a better template to integrate it into another project in some cases.

* Most user-definable types are dummies. Nodes and edges do not carry any payload. There is only one node type.
* Persistence is removed.
* Web support is removed.
* The example showcases how to extract graph structure from `state.graph`.
* The example also demonstrates how to add a node from out outside (without using a built-in node creation menu).

I expect removed features should be relatively straightforward to re-add by looking at the main example.

kamirr commented:
Thanks for the PR! I like the idea, but I think an introductory example should have some comments explaining what's doing what. If you could address that as well as the CI failures I'll merge this. Cheers!

Edit: Also this PR needs to modify README.md to mention the new example. :)

vi commented:
> should have some comments explaining what's doing what. 

Composed some comments (based on my understanding). Please check whether they are accurate.

> CI failures

Fixed the `fmt` and clippies in new code, but CI shows clippies in `egui-graph-edit/src` a well. Did not touch that.

> README.md to mention the new example

Added a link.

kamirr commented:
I've merged main into this PR and updated egui to 0.31.0 (happened in parallel on main).
